src/components/views/ViewMilestone.jsx

```diff
@@ -121,27 +137,26 @@ const ViewMilestone = props => {
         if (milestoneId) {
           history.push(`/milestone/${_milestone.slug}`);
         }
-        _milestoneId = _milestone.id;
         setMilestone(_milestone);
         setCampaign(new Campaign(_milestone.campaign));
         setRecipient(
           _milestone.pendingRecipientAddress ? _milestone.pendingRecipient : _milestone.recipient,
         );
         getDacTitle(_milestone.dacId);
-        loadMoreDonations();
+        donationsObserver = MilestoneService.subscribeNewDonations(
```

Comment:
We may need to subscription types for milestone, one for any non failed donations and one for not failed and not pending maybe